demo/dashboard/client/src/pages/files/file/Logs.js

```diff
@@ -0,0 +1,142 @@
+/* eslint-disable no-underscore-dangle */
+
+import React from 'react';
+import elasticsearch from 'elasticsearch';
+import PropTypes from 'prop-types';
+import Spinner from '../../../components/Spinner';
+
+const logsApiEndpoint = `https://${process.env.REACT_APP_ES_LOGS_ENDPOINT}`;
+const logsIndex = `${process.env.REACT_APP_STAGE}-logs`;
+
+class Logs extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      logs: [],
+      logsLoading: false,
+      logsCount: 0,
+    };
+
+    this.loadLogs = this.loadLogs.bind(this);
+    this.setLogs = this.setLogs.bind(this);
+    this.setEmptyLogs = this.setEmptyLogs.bind(this);
+  }
+
+  componentDidMount() {
+    this.logClient = elasticsearch.Client({
+      host: logsApiEndpoint,
+      apiVersion: '5.5',
+      log: 'warning',
+    });
+
+    this.loadLogs();
+  }
+
+  // Load related logs
+  loadLogs() {
+    const { computedKey } = this.props;
+
+    this.setState({ logsLoading: true }, this.setLogs(computedKey));
+  }
+
+  setLogs(computedKey) {
+    return () =>
+      this.logClient.indices
```

Comment:
It's ok to leave it for now, though at some point, let's think of getting the ES client logic out in a helper or a lib, out of the React components so we can keep the view layer simpler to understand and maintain?

Reply:
There's always place for improvement 😄 I agree with you on that point, we should extract the logic and make the views great again!